cirrus/server/cirrus/main.py

```diff
@@ -76,9 +82,22 @@ def read_root():
 
 
 @app.post("/v1/features/", status_code=status.HTTP_200_OK)
-async def compute_features():
-    # # will recieve as part of incoming request
-    targeting_context: Dict[str, Any] = {"clientId": "test", "requestContext": {}}
+async def compute_features(request_data: FeatureRequest):
+    if not request_data.client_id:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Client ID value is missing or empty",
+        )
+
+    if not request_data.context:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Context value is missing or empty",
+        )
```

Comment:
I see, so if the key is missing you get 422 but if the key is present but null or context is `{}` or anything falsy, you get 400.  Interesting.  I wonder if these shoudl all be 422s just to be consistent?  Just a thought, no need to change it right now.

Reply:
yeah I was debating myself on this should it be 400 vs 422, but good point to be consistent, I will keep a note of these and can change it later 💪